modules/core/src/main/scala/Span.scala

```diff
@@ -21,4 +25,9 @@ trait Span[F[_]] {
   /** Resource that yields a child span with the given name. */
   def span(name: String): Resource[F, Span[F]]
 
+  def mapK[G[_]](fk: F ~> G)(implicit b: Bracket[F, Throwable], deferG: Defer[G], applicativeg: Applicative[G]): Span[G] = new Span[G] {
+    def put(fields: (String, TraceValue)*): G[Unit] = fk(self.put(fields:_*))
```

Comment:
I think `put` could take a List of fields instead, with a varargs overload implemented as an alias...